blockprod/src/detail/mod.rs

```diff
@@ -180,19 +180,51 @@ impl BlockProduction {
         &self,
         current_tip: Id<GenBlock>,
         min_block_timestamp: BlockTimestamp,
+        transactions: Vec<SignedTransaction>,
+        transaction_ids: Vec<Id<Transaction>>,
+        include_mempool: bool,
     ) -> Result<Option<Box<dyn TransactionAccumulator>>, BlockProductionError> {
-        let max_block_size = self.chain_config.max_block_size_from_std_scripts();
+        let mut accumulator: Box<dyn TransactionAccumulator + Send> =
+            Box::new(DefaultTxAccumulator::new(
+                self.chain_config.max_block_size_from_std_scripts(),
+                current_tip,
+                min_block_timestamp,
+            ));
+
+        for transaction in transactions.into_iter() {
+            let transaction_id = transaction.transaction().get_id();
+            // TODO: fees
+            accumulator
+                .add_tx(transaction, Amount::ZERO.into())
+                .map_err(|err| BlockProductionError::FailedToAddTransaction(transaction_id, err))?
+        }
+
         let returned_accumulator = self
             .mempool_handle
-            .call(move |mempool| {
-                mempool.collect_txs(Box::new(DefaultTxAccumulator::new(
-                    max_block_size,
-                    current_tip,
-                    min_block_timestamp,
-                )))
+            .call({
+                move |this| -> Result<_, BlockProductionError> {
+                    for transaction_id in transaction_ids.iter() {
+                        if let Some(transaction) = this.transaction(transaction_id) {
+                            // TODO: fees
+                            accumulator.add_tx(transaction, Amount::ZERO.into()).map_err(|err| {
+                                BlockProductionError::FailedToAddTransaction(*transaction_id, err)
+                            })?
+                        } else {
+                            // TODO: shoud this error out rather than ignore missing transactions?
+                        }
```

Comment:
Wasn't too sure if this should just ignore missing transactions, or this actually is an error.